activerecord/lib/arel/nodes/comment.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Arel # :nodoc: all
+  module Nodes
+    class Comment < Arel::Nodes::Node
+      SQL_COMMENT_DELIMITER_PATTERN = /\/\*|\*\//.freeze
+
+      attr_reader :value
+
+      def initialize(value)
+        super()
+        value = sanitize_sql_comment_delimiters(value.to_s)
+        @value = Arel.sql(value)
+      end
+
+      def initialize_copy(other)
+        super
+        @value = @value.clone if @value
+      end
+
+      def hash
+        [@value].hash
+      end
+
+      def eql?(other)
+        self.class == other.class &&
+          self.value == other.value
+      end
+      alias :== :eql?
+
+      private
+
+        def sanitize_sql_comment_delimiters(str)
+          while str.match?(SQL_COMMENT_DELIMITER_PATTERN)
+            str = str.gsub(SQL_COMMENT_DELIMITER_PATTERN, "")
+          end
+          str
+        end
```

Comment:
Should we catch magic comment injection here as well, e.g. `/*! … ` extensions in MySQL or `/*+` optimizer hints?

We do prevent these cases due to the SQL visitor emitting spaces around the value. Would we notice the injection if it stopped emitting spaces? (And whose responsibility is it to recognize that: the SQL visitor or the node?)